pymc4/flow/executor.py

```diff
@@ -118,29 +123,31 @@ def evaluate_model(self, model, state=None):
                 break
         return return_value, state
 
+    __call__ = evaluate_model
+
 
 SamplingState = collections.namedtuple("EvaluationState", "values,distributions,potentials")
 
 
 class SamplingExecutor(Executor):
-    def __init__(self, **do):
-        self.default_do = do
+    def __init__(self, *do: dict, **do_kwargs: dict):
```

Comment:
what does `do` mean? the `do` operator, conditioning?

Reply:
Exactly

Reply:
I'm not sure that's right here, do != conditioning as I understand it. do means if I were to fix a value and the causal links work correctly (i.e. fixing the thermometer to a value does not change conditional of the temperature). Plain conditioning does not necessarily incorporate causality (i.e. the conditional distribution of the temperature is dependent on the thermometer value). I would just call it conditioning here, maybe `fix`, `cond`, `condition`, `depends`, while I appreciate the succinctness of `do`.

Reply:
The way it appears in the graph, this is exactly a `do` operator. In case you have the following dependence and state provided for 
```
do = {"a": 1}
graph:
    b->a->c
```
You come up with
```
    b->do(a, inserted from state)->c
```

Reply:
However, for the leaf nodes, the observed, do=conditioning

Reply:
My fault, the logp computation seems to work as conditioning, you are right